### PR TITLE
Add service annotations support to Trino Gateway Chart

### PR DESCRIPTION
**Pull request by yashsinghalso**
- Introduced functionality to allow specifying annotations for Trino Gateway services.
- Ensures compatibility with Kubernetes custom annotations for enhanced service configurations.
- Updated relevant templates and documentation to reflect the changes.

This feature enables users to add annotations for service monitoring, ingress, or other Kubernetes integrations.

### CODE REVIEW

File: charts/gateway/templates/_helpers.tpl
Comment (nineinchnick) [inline charts/gateway/templates/_helpers.tpl:74]: Please keep the end of line (EOL) character at the end of file (EOF).

File: charts/gateway/templates/_helpers.tpl
Comment (nineinchnick) [inline charts/gateway/templates/_helpers.tpl:70]: What's the purpose of this template? We should expect correct values to be provided in the `valeus.yaml` file.

Comment (yashsinghalso) [reply charts/gateway/templates/_helpers.tpl]: Have removed the template.

**Comment by nineinchnick** [comment]
@cla-bot check

**Comment by yashsinghalso** [comment]
@cla-bot check

**Review by nineinchnick — APPROVED** [review]
LGTM, please squash commits and remove the `feat:` prefix

**Comment by yashsinghalso** [comment]
Can you provide me the write access? The merging is blocked for me- _due to failing merge requirements._

**Comment by nineinchnick** [comment]
I'll merge this PR once you squash the commits locally and push it to this branch. It's more important to have a good commit message, that's why I'm asking you to address this. See the end of this section: https://trino.io/development/process#pull-request-and-commit-guidelines-

**Pull request merged**
